### PR TITLE
refactor(ivy): update TNodeType

### PR DESCRIPTION
**Pull request by benlesh**
- Remove an extra type `ViewOrElement`, which even had the same numeric value as `View`.
- Updates comment to remove part about alleged bit-masking that we could be doing here.
  We aren't using this with bitmasks, and if we were, everything would be a `NodeType.Container`,
  because its value was `0`.
- Updates the number values to be simple, human-readable integers, since we're not using these
  with any kind of bit-manipulation.
- Add comments about each type.

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]
LGTM!

**Comment by benlesh** [comment]
[presubmit](http://test/OCL:238703994:BASE:238704083:1552683092709:ded1c07a)

File: packages/core/src/render3/interfaces/node.ts [hunk 1]
```diff
@@ -12,17 +12,33 @@ import {LView, TView} from './view';
...
+  /**
+   * The TNode contains information about an {@link LView}
+   */
+  View = 2,
+  /**
+   * The TNode contains information about a DOM element aka {@link RNode}.
+   */
+  Element = 3,
+  /**
+   * The TNode contains information about a container comment {@link RNode}.
```
Comment (kara) [inline packages/core/src/render3/interfaces/node.ts]: I would say ng-container somewhere in here, since the comment is an implementation detail

**Review by kara — CHANGES_REQUESTED** [review]
One small thing

**Review by kara — APPROVED** [review]
LGTM

**Comment by benlesh** [comment]
Caretaker: CI flakes with this PR.

**Pull request closed without merging**